src/ucs/sys/sock.c

```diff
@@ -321,8 +328,7 @@ ucs_socket_handle_io_error(int fd, const char *name, ssize_t io_retval, int io_e
     }
 
     if (err_cb != NULL) {
-        status = err_cb(err_cb_arg, io_errno);
-        if (status == UCS_OK) {
+        if (err_cb(err_cb_arg, io_errno) == UCS_OK) {
             return UCS_ERR_NO_PROGRESS;
```

Comment:
if UCS_ERR_NOT_CONNECTED returns from ucs_socket_check_errno(), will this status be lost in line 332 when returning UCS_ERR_NO_PROGRESS?

Reply:
yes, in case of user's callback return UCS_OK we override this status to ERR_NO_PROGRESS to avoid printing errors in case of when it is expected
see description for this err callback:
```
/*
  ... Returns an error if the default error handling should be
 * done, otherwise - UCS_OK */
typedef ucs_status_t (*ucs_socket_io_err_cb_t)(void *arg, int io_errno);
```
I fixed the comment for this callback

Reply:
so if the remote peer disconnected and the local error handler was invoked, the caller of the send_nb function won't have the UCS_ERR_NOT_CONNECTED returned to him?

Reply:
> so if the remote peer disconnected and the local error handler was invoked, the caller of the send_nb function won't have the UCS_ERR_NOT_CONNECTED returned to him?

yes, it may not want to ucs_error some error - e,g, this is a case when UCT/TCP resolving simultaneous connection of two EPs to each other. An EP that has lower IP addr/port closes socket fd, so remote peer can be in a state which requires to receive smth from the peer - in order to avoid printing errors in such situation, we disable it by returning UCS_OK from user's error callback